enclave-runner/src/usercalls/mod.rs

```diff
@@ -693,61 +817,44 @@ impl EnclaveState {
         };
         let enclave_clone = enclave.clone();
         let io_future = async move {
+            let (usercall_queue_synchronizer, return_queue_synchronizer, sync_usercall_tx) = QueueSynchronizer::new(enclave_clone.clone());
+
+            let (usercall_queue_tx, usercall_queue_rx) = ipc_queue::bounded_async(USERCALL_QUEUE_SIZE, usercall_queue_synchronizer);
+            let (return_queue_tx, return_queue_rx) = ipc_queue::bounded_async(RETURN_QUEUE_SIZE, return_queue_synchronizer);
+
+            let fifo_guards = FifoGuards {
+                usercall_queue: usercall_queue_tx.into_descriptor_guard(),
+                return_queue: return_queue_rx.into_descriptor_guard(),
+                async_queues_called: false,
+            };
+
+            *enclave_clone.fifo_guards.lock().await = Some(fifo_guards);
+            *enclave_clone.return_queue_tx.lock().await = Some(return_queue_tx);
+
+            tokio::task::spawn_local(async move {
+                while let Ok(usercall) = usercall_queue_rx.recv().await {
+                    let _ = io_queue_send.send(UsercallSendData::Async(usercall));
+                }
+            });
```

Comment:
That's possible, but it would complicate the already too complicated loop below. What's your main concern with having it as a separate task?

Reply:
Primarily because the current flow is obfuscating what the input is to the while loop below and a lesser concern might be performance.

Reply:
The input to that while loop comes from `fn do_work` and this task, I don't see that being obfuscated, specially because `UsercallSendData` has two variants. I don't think performance will be impacted much by having an extra task considering that doing `select` will also introduce an extra future.

We could also implement some form of back pressure on the usercall queue more easily in this form, but that might not be needed in practice.